src/core/analysis/rnn_id_resolver.h

```diff
@@ -73,7 +73,8 @@ struct RnnCrdHasher {
     v ^= 0xfeadbeed1235;
     v *= 0x12312f12aff1ULL;
     v ^= (v >> 33);
-    return v;
+    //TODO: Can algorithm support u32?
+    return static_cast<size_t>(v);
```

Comment:
I coded this with an assumption that sizeof(size_t) == 8, so this is not optimal on 32 bits.

Reply:
This is basically a type punning (read struct as a single u64).
Should make sense if sizeof(RnnCoordinate) == sizeof(u64).
There could be endianess problems, but there should not be UB here. memcpy can be used as C++20 std::bitcast.

Reply:
Yeah I understand about assumption, but hashers should return `size_t` so I'm not sure what we can do with it (because if I'm not mistaken that what is expected in `std::hash`)

Reply:
It should be safe to drop upper 32 bits here, hashing function mixes upper bits (that have higher randomness with LCG) with lower ones.
But we definitely need to read all 64 bits of original for hash to work.
So let's go with your fixes.